### PR TITLE
CMI 8338/8738 fixes

### PR DESCRIPTION
**Pull request by jiyunomegami**
Tested with a card with the CMI8738/PCI-6ch-MX chip

### CODE REVIEW

File: mpxplay/au_cards/sc_cmi.c [hunk 1]
```diff
@@ -745,19 +775,33 @@ static long CMI8X38_getbufpos(struct mpxplay_audioout_info_s *aui)
  struct cmi8x38_card *card=aui->card_private_data;
  unsigned long bufpos;
 
+#if 0
```
Comment (crazii) [inline mpxplay/au_cards/sc_cmi.c:778]: use #ifndef SBEMU will be better, that will keep the original code unchanged for non-SBEMU usage.

**Comment by crazii** [comment]
Thanks!
I was still check the code for ES1371. Very much appreciated if you happen to have that card and want help.

**Comment by crazii** [comment]
The guard of SBEMU will keep the original code unmodified, and also will help the author of mpxplay to review code if needed.

EDIT: Your code also helps me to fix the ES1371 problems.

File: mpxplay/au_cards/sc_cmi.c (hunk 1 above)
Comment (jiyunomegami) [reply mpxplay/au_cards/sc_cmi.c:778]: Hold on, I will just delete that stuff.

**Comment by jiyunomegami** [comment]
The changes in CMI8X38_getbufpos are just to make it work like the Linux driver, they have nothing to do with mpxplay.
The Linux driver tries to read the PCM pointer 3 times, and adds 1 to it in the calculation of bufpos.

**Comment by jiyunomegami** [comment]
I added another commit with the #if 0/#if 1 stuff removed. Tested with Monkey Island and Duke Nukem 3D.

**Comment by crazii** [comment]
OK, can you keep the last deleted block of code? I can do that after merge if your time is limited.
BTW, does it happens that the remained sample count exceed the dma_size (rem>card->dma_size)?  is it a bug of the soundcard? Is there any comments in Linux driver concerning that?

**Comment by jiyunomegami** [comment]
I don't have an ES1371, but I tested cards with these 3 chips:
CMI8738/PCI-6ch-MX MT5L3.03-1051-GS
CMI8738/PCI M9J9F-037D
CMI8738/PCI-6ch-LX (Aopen AW850 Cobra, like the one in the PhilsComputerLab video)

**Comment by crazii** [comment]
Sorry I cannot reply in time due to the firewall

**Comment by jiyunomegami** [comment]
>OK, can you keep the last deleted block of code?

Please do that after merge, since I am not sure what you mean.


You can see the Linux driver here: https://github.com/torvalds/linux/blob/master/sound/pci/cmipci.c
The only comment, in the function snd_cmipci_pcm_pointer, is "this seems better.." LOL

**Comment by crazii** [comment]
OK, cool, thanks!

**Pull request merged**